src/otx/cli/utils/workspace.py

```diff
@@ -0,0 +1,33 @@
+"""Class modules that manage Workspace."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from datetime import datetime, timezone
+from pathlib import Path
+
+
+class Workspace:
+    """Represents a workspace for the OTX application.
+
+    Args:
+        work_dir (Path | str | None, optional): The path to the workspace directory. Defaults to None.
+        use_sub_dir (bool, optional): Whether to use a subdirectory within the workspace. Defaults to True.
+    """
+
+    def __init__(self, work_dir: Path | str | None = None, use_sub_dir: bool = True):
+        if work_dir is None:
+            if not (Path.cwd() / ".cache").exists():
+                # Without work_dir input & no .cache directory in root
+                self.work_dir = Path.cwd() / "otx-workspace"
+            else:
+                # If Path.cwd is workspace
+                self.work_dir = Path.cwd()
+        else:
+            self.work_dir = Path(work_dir)
```

Comment:
```suggestion
    def __init__(self, work_dir: Path = Path.cwd(), use_sub_dir: bool = True):
            self.work_dir = (
                work_dir / "otx-workspace"
                # Without work_dir input & no .cache directory in root
                if work_dir == Path.cwd() and not (work_dir / ".cache").exists()            
                else work_dir
            )
```

I'm feeling a little bit tired to say this kind of comment again and again https://github.com/openvinotoolkit/training_extensions/pull/2912#discussion_r1491842254.
What's the reason that this happens again and again?

Reply:
I didn't pay attention to the other reviewer's comment in the other review. Anyway, I've fixed it, thanks. - https://github.com/openvinotoolkit/training_extensions/pull/2934/commits/6b560edf42ebdfdbf9bc3268df6adcb375a47f07